lib/gtfs_realtime_viz.ex

```diff
@@ -36,16 +36,36 @@ defmodule GTFSRealtimeViz do
   Renders the received protobuf files and comments into an HTML fragment that can either
   be opened directly in a browser or embedded within the HTML layout of another app.
   """
-  @spec visualize(term) :: String.t
-  def visualize(group) do
-    [vehicle_archive: vehicles_by_stop_id(group), routes: routes()]
+  @spec visualize(term, %{String.t => [[String.t]]}) :: String.t
+  def visualize(group, opts) do
+    all_locations = locations_we_care_about(opts)
+    vehicles_we_care_about = group
+                             |> State.vehicles
+                             |> vehicles_we_care_about(all_locations)
+
+    [vehicle_archive: vehicles_by_stop_id(vehicles_we_care_about), routes: opts]
     |> gen_html
     |> Phoenix.HTML.safe_to_string
   end
 
-  @spec vehicles_by_stop_id(term) :: [{String.t, %{required(String.t) => [Proto.vehicle_position]}}]
-  defp vehicles_by_stop_id(group) do
-    Enum.map(State.vehicles(group), fn {comment, vehicles} ->
+  def locations_we_care_about(opts) do
```

Comment:
Ah, so we actually care about the routes, not the stops, though I suppose the ticket is not clear on that. Sorry! The use case that triggered it was wanting to view, say, just the ladder chart for vehicles on Green-E; filtering by stop like this would include vehicles on Green-B/C/D, that are currently at the shared stops.

So with that, I think this function can be `opts |> Map.keys |> MapSet.new` and other code would check `position.trip && MapSet.member?(routes, position.trip.route_id)`.

Reply:
i suppose thats true - is it functionally different though?  you would still be passing the list of locations on the routes.  Maybe your way allows you to just pass %{"route_name" => []} when you dont care about namign the stops?